app/services/assignment_service.py

```diff
@@ -177,6 +182,9 @@ async def update_assignment(self, assignment: AssignmentModel, update_assignment
         if "due_date" in update_fields:
             assignment.due_date = update_fields["due_date"]
 
+        if "is_published" in update_fields:
+            assignment.is_published = update_fields["is_published"]
```

Comment:
you need to add this to both UpdateAssignmentSchema and the schema used for the API endpoint handler that updates assignments.

Reply:
Yep good catch